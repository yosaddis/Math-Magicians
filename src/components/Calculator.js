import React, { Component } from 'react';
import './Calculator.css';

export default class Calculator extends Component {
  render() {
    return (
      <div className="main-container">
        <div className="result-display" />
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-zero">0</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-dot">.</button>
        <button type="button" onClick={this.btnOnClickResult} className="btn-ops btn-equal">=</button>
        <button type="button" onClick={this.btnOnClick} className="btn-light btn-clear">AC</button>
        <button type="button" onClick={this.btnOnClick} className="btn-light btn-toogle-sign">+/-</button>
        <button type="button" onClick={this.btnOnClick} className="btn-light btn-percentage">%</button>
        <button type="button" onClick={this.btnOnClick} className="btn-ops btn-divider">/</button>
        <button type="button" onClick={this.btnOnClick} className="btn-ops btn-multiply">*</button>
        <button type="button" onClick={this.btnOnClick} className="btn-ops btn-substract">-</button>
        <button type="button" onClick={this.btnOnClick} className="btn-ops btn-plus">+</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-one">1</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-two">2</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-three">3</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-four">4</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-five">5</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-six">6</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-seven">7</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-eight">8</button>
        <button type="button" onClick={this.btnOnClickOperator} className="btn-light btn-nine">9</button>
      </div>
    );
  }
}
