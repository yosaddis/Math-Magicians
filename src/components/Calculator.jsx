import React, { Component } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import Button from './ButtonActions';

export default class Calculator extends Component {
  // constructor
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      next: null,
      opertation: null,
    };
    this.btnOnClick = this.btnOnClick.bind(this);
  }

  // action listner for button object
  btnOnClick(buttonName) {
    this.setState((prevState) => (calculate(prevState, buttonName)));
  }

  // displaying the buttons in arrangement from css
  render() {
    const { total, next, opertation } = this.state;
    return (
      <div className="main-container">
        <div className="result-display">
          {total || ''}
          {opertation || ''}
          {next || ''}
        </div>
        <Button typeBtn="0" btnOnClick={this.btnOnClick} btnClass="btn-light btn-zero" />
        <Button typeBtn="." btnOnClick={this.btnOnClick} btnClass="btn-light btn-dot" />
        <Button typeBtn="=" btnOnClick={this.btnOnClick} btnClass="btn-ops btn-equal" />
        <Button typeBtn="AC" btnOnClick={this.btnOnClick} btnClass="btn-light btn-clear" />
        <Button typeBtn="+/-" btnOnClick={this.btnOnClick} btnClass="btn-light btn-toogle-sign" />
        <Button typeBtn="%" btnOnClick={this.btnOnClick} btnClass="btn-light btn-percentage" />
        <Button typeBtn="/" btnOnClick={this.btnOnClick} btnClass="btn-ops btn-divider" />
        <Button typeBtn="x" btnOnClick={this.btnOnClick} btnClass="btn-ops btn-multiply" />
        <Button typeBtn="-" btnOnClick={this.btnOnClick} btnClass="btn-ops btn-substract" />
        <Button typeBtn="+" btnOnClick={this.btnOnClick} btnClass="btn-ops btn-plus" />
        <Button typeBtn="1" btnOnClick={this.btnOnClick} btnClass="btn-light btn-one" />
        <Button typeBtn="2" btnOnClick={this.btnOnClick} btnClass="btn-light btn-two" />
        <Button typeBtn="3" btnOnClick={this.btnOnClick} btnClass="btn-light btn-three" />
        <Button typeBtn="4" btnOnClick={this.btnOnClick} btnClass="btn-light btn-four" />
        <Button typeBtn="5" btnOnClick={this.btnOnClick} btnClass="btn-light btn-five" />
        <Button typeBtn="6" btnOnClick={this.btnOnClick} btnClass="btn-light btn-six" />
        <Button typeBtn="7" btnOnClick={this.btnOnClick} btnClass="btn-light btn-seven" />
        <Button typeBtn="8" btnOnClick={this.btnOnClick} btnClass="btn-light btn-eight" />
        <Button typeBtn="9" btnOnClick={this.btnOnClick} btnClass="btn-light btn-nine" />
      </div>
    );
  }
}
