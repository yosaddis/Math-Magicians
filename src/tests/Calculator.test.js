import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import Calculator from '../components/Calculator';

describe('Unit test for calculator component', () => {
  test('All Buttons should be rendered correctly', () => {
    render(<Calculator />);
    expect(screen.getAllByRole('button').length).toBe(19);
  });
  test('The display component inside calculator should be rendered correctly', () => {
    render(<Calculator />);
    expect(screen.getByTestId('result-display')).toBeInTheDocument();
  });

  test('The calculator snap shot should be rendered correctly', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
  });
});
