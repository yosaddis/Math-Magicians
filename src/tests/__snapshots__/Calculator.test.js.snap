// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for calculator component The calculator snap shot should be rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-container"
      >
        <div
          class="result-display"
          data-testid="result-display"
        >
          0
        </div>
        <button
          class="btn-light btn-zero"
          type="button"
        >
          0
        </button>
        <button
          class="btn-light btn-dot"
          type="button"
        >
          .
        </button>
        <button
          class="btn-ops btn-equal"
          type="button"
        >
          =
        </button>
        <button
          class="btn-light btn-clear"
          type="button"
        >
          AC
        </button>
        <button
          class="btn-light btn-toogle-sign"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn-light btn-percentage"
          type="button"
        >
          %
        </button>
        <button
          class="btn-ops btn-divider"
          type="button"
        >
          /
        </button>
        <button
          class="btn-ops btn-multiply"
          type="button"
        >
          x
        </button>
        <button
          class="btn-ops btn-substract"
          type="button"
        >
          -
        </button>
        <button
          class="btn-ops btn-plus"
          type="button"
        >
          +
        </button>
        <button
          class="btn-light btn-one"
          type="button"
        >
          1
        </button>
        <button
          class="btn-light btn-two"
          type="button"
        >
          2
        </button>
        <button
          class="btn-light btn-three"
          type="button"
        >
          3
        </button>
        <button
          class="btn-light btn-four"
          type="button"
        >
          4
        </button>
        <button
          class="btn-light btn-five"
          type="button"
        >
          5
        </button>
        <button
          class="btn-light btn-six"
          type="button"
        >
          6
        </button>
        <button
          class="btn-light btn-seven"
          type="button"
        >
          7
        </button>
        <button
          class="btn-light btn-eight"
          type="button"
        >
          8
        </button>
        <button
          class="btn-light btn-nine"
          type="button"
        >
          9
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-container"
    >
      <div
        class="result-display"
        data-testid="result-display"
      >
        0
      </div>
      <button
        class="btn-light btn-zero"
        type="button"
      >
        0
      </button>
      <button
        class="btn-light btn-dot"
        type="button"
      >
        .
      </button>
      <button
        class="btn-ops btn-equal"
        type="button"
      >
        =
      </button>
      <button
        class="btn-light btn-clear"
        type="button"
      >
        AC
      </button>
      <button
        class="btn-light btn-toogle-sign"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn-light btn-percentage"
        type="button"
      >
        %
      </button>
      <button
        class="btn-ops btn-divider"
        type="button"
      >
        /
      </button>
      <button
        class="btn-ops btn-multiply"
        type="button"
      >
        x
      </button>
      <button
        class="btn-ops btn-substract"
        type="button"
      >
        -
      </button>
      <button
        class="btn-ops btn-plus"
        type="button"
      >
        +
      </button>
      <button
        class="btn-light btn-one"
        type="button"
      >
        1
      </button>
      <button
        class="btn-light btn-two"
        type="button"
      >
        2
      </button>
      <button
        class="btn-light btn-three"
        type="button"
      >
        3
      </button>
      <button
        class="btn-light btn-four"
        type="button"
      >
        4
      </button>
      <button
        class="btn-light btn-five"
        type="button"
      >
        5
      </button>
      <button
        class="btn-light btn-six"
        type="button"
      >
        6
      </button>
      <button
        class="btn-light btn-seven"
        type="button"
      >
        7
      </button>
      <button
        class="btn-light btn-eight"
        type="button"
      >
        8
      </button>
      <button
        class="btn-light btn-nine"
        type="button"
      >
        9
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
